ccdproc/image_collection.py

```diff
@@ -271,7 +284,13 @@ def refresh(self):
         """
         keywords = '*' if self._all_keywords else self.keywords
         # Re-load list of files
-        self._files = self._fits_files_in_directory()
+        if self._filenames:
```

Comment:
This seems exactly like the block in `__init__`. To ease maintenance could you create a helper-method or check if this block could be put into `_fits_files_in_directory`?